enhancements/network/cloud-egress-ip.md

```diff
@@ -0,0 +1,445 @@
+---
+title: TBD - for now: Cloud API component for egress IP
+authors:
+  - "@alexanderConstantinescu"
+reviewers:
+  - "@danwinship"
+  - "@abhat"
+  - "@squeed"
+  - "@tssurya"
+  - "@trozet"
+approvers:
+  - TBD
+creation-date: 2020-12-16
+last-updated: -
+status: implementable
+---
+
+# TBD - for now: Cloud API component for egress IP
+
+This covers adding cloud provider support for the OpenShift networking feature:
+egress IP. It is not clear today what the extent of the component presented here
+will be, see: [Summary](#summary). I am thus leaving the title open-ended for
+further refinement once people review this enhancement proposal.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+OVN-Kubernetes and openshift-sdn today support the feature egress IP on
+bare-metal platforms. Going forwards OpenShift will need to support the feature
+on conventional cloud provider platforms (such as: GCP, AWS, Azure). Cloud
+providers require additional management of the egress IP address via the cloud
+provider's cloud API. That is required since the cloud provider will not let the
+network plugin claim an additional IP on the node, which it does not recognize.
+
+To be able to do this we will need to add logic (to either a dedicated component
+or to the network plugins themselves) as to be able to handle this. This
+enhancement proposal concerns itself with a discussion surrounding questions
+such as:
+
+- If the component should be generic enough as to allow other resources the
```

Comment:
:+1: I think whatever design we come up with should account for the capability to talk to cloud API to modify host/cluster configurations such as security groups, other gateway configurations etc.

Reply:
I don't think we necessarily need a single _component_ that can do all of these things. We would of course want to be able to reuse some of the code at least, but (assuming we even ever implement those other features), it might end up making more sense to integrate them elsewhere. (Especially, we might decide it makes more sense to implement the egress IP part separately because it could also be used by non-OCP users of ovn-kubernetes.)

Reply:
Yeah, I agree with @danwinship . I think it's better to design for things we know we'll implement, hence: only egress IP. Most things in this design proposal is already very vague, making it complex enough.